File: C++/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.cpp
#include <string>
#include <vector>

using namespace std;

int solution(int num) {
    int i = 0;
    long n = num;
    for(i = 0; i < 500; i++) {
        if (n == 1) break;
        if(n % 2 == 0) {
            n /= 2;
        } else {
            n = n * 3 + 1;
        }
    }
    return i == 500 ? -1 : i;
}

File: Python3/프로그래머스/1/12943. 콜라츠 추측/콜라츠 추측.py
def solution(num):
    answer = 0
    while num != 1 and answer < 500:
        answer += 1
        if num % 2 == 0:
            num /= 2
            num = int(num)
        else:
            num = num * 3 + 1
        
    return answer if answer != 500 else -1